src/components/layouts/MainLayout.tsx

```diff
@@ -1,33 +1,85 @@
 'use client'
 
-import * as React from 'react'
-import Header from './Header'
-import Sidebar from './Sidebar'
-import { useState } from 'react'
+import * as React from 'react';
+import Header from './Header';
+import Sidebar from './Sidebar';
+import TrendingSidebar from './TrendingSidebar';
+import { trendingPostsData, trendingPeopleData } from '@/utils/constant/trending-post/index';
+import ButtonOption from '@/components/trending-post/button-option';
+import { buttonOptions } from '@/utils/constant/buttonOptions';
+import { useState, useEffect } from 'react';
 
 interface MainLayoutProps {
-  children: React.ReactNode
-  title: string
+  children: React.ReactNode;
+  title: string;
 }
 
 const MainLayout = ({ children, title }: MainLayoutProps) => {
-  const [isSidebarOpen, setIsSidebarOpen] = useState(false)
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const [isMobileView, setIsMobileView] = useState(false);
+  const [isTrendingSidebarOpen, setIsTrendingSidebarOpen] = useState(false);
+  const [selectedButtonIndex, setSelectedButtonIndex] = useState(-1);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobileView(window.innerWidth < 768);
+    };
+
+    window.addEventListener('resize', handleResize);
+    handleResize();
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
 
   const toggleSidebar = () => {
-    setIsSidebarOpen(!isSidebarOpen)
-  }
+    setIsSidebarOpen(!isSidebarOpen);
+  };
+
+  const handleButtonOptionClick = (index: number) => {
+    if (index === buttonOptions.length - 1 && isMobileView) {
+      setIsTrendingSidebarOpen(!isTrendingSidebarOpen);
+      setSelectedButtonIndex(isTrendingSidebarOpen ? -1 : index);
+    } else {
+      setSelectedButtonIndex(index);
+      setIsTrendingSidebarOpen(false);
+    }
+  };
 
   return (
-    <div>
+    <div className="relative min-h-screen flex">
       <Sidebar isSidebarOpen={isSidebarOpen} toggleSidebar={toggleSidebar} />
-      <Header
-        title={title}
-        isSidebarOpen={isSidebarOpen}
-        toggleSidebar={toggleSidebar}
+      <div className={`flex-1 flex flex-col transition-transform duration-300 ease-in-out ${isSidebarOpen ? 'ml-[300px]' : ''}`}>
+        <Header
+          title={title}
+          isSidebarOpen={isSidebarOpen}
+          toggleSidebar={toggleSidebar}
+        />
+        <main className="flex-1">
+          {children}
+        </main>
+      </div>
+      <TrendingSidebar
+        posts={trendingPostsData}
+        trendingPeople={trendingPeopleData}
+        buttons={buttonOptions}
+        isVisible={isTrendingSidebarOpen && isMobileView} 
       />
-      <div>{children}</div>
+      {isMobileView && (
```

Comment:
check MobileView dùng tailwind là được mà. muốn hidden khi lên màn to thì dùng md:hidden là được